sdk/servicebus/service-bus/test/internal/batchingReceiver.spec.ts

```diff
@@ -116,4 +127,153 @@ describe("BatchingReceiver unit tests", () => {
       assert.isEmpty(callsDoneAfterAbort);
     });
   });
+
+  /**
+   * receive(max messages, max wait time, max wait time past first message) has 3 exit paths:
+   * 1. We received 'max messages'
+   * 2. We've waited 'max wait time'
+   * 3. We've received 1 message and _now_ have exceeded 'max wait time past first message'
+   */
+  [ReceiveMode.peekLock, ReceiveMode.receiveAndDelete].forEach((lockMode) => {
+    describe(`${ReceiveMode[lockMode]} receive, exit paths`, () => {
+      const bigTimeout = 60 * 1000;
+      const littleTimeout = 30 * 1000;
+      let clock: ReturnType<typeof sinon.useFakeTimers>;
+
+      beforeEach(() => {
+        clock = sinon.useFakeTimers();
+      });
+
+      afterEach(() => {
+        clock.restore();
```

Comment:
:+1:

Reply:
IT'S SO AWESOME. THE POWER TO CONTROL TIME!